pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,352 @@
+        self.currentBrush = self.brush
+
+        self._shape = None
+
+        self._pos = (0, 0)
+        if pos is None:
+            pos = (0, 0)
+        self.setPos(pos)
+
+        if isinstance(symbol, str):
+            try:
+                self._path = Symbols[symbol]
+            except KeyError:
+                raise KeyError("symbol name found in available Symbols")
+        elif isinstance(symbol, QtGui.QPainterPath):
+            self._path = symbol
+        else:
+            raise TypeError("Unknown type provided as symbol")
+        
+
+        self.scale = diameter
+        self.setPath(self._path)
+        self.setLabel(label, labelOpts)
+
+    def setPos(self, pos):
+        if isinstance(pos, tuple):
+            newPos = pos
+        elif isinstance(pos, list):
+            newPos = tuple(pos)
+        elif isinstance(pos, (QtCore.QPointF, QtCore.QPoint)):
+            newPos = (pos.x(), pos.y())
+        else:
+            raise TypeError
+        if self._pos != newPos:
+            self._pos = newPos
+            super().setPos(Point(self._pos))
+            self.sigPositionChanged.emit(self)
+
+    def setBrush(self, *args, **kwargs):
+        """Set the brush that fills the cursor. Allowable arguments are any that
+        are valid for :func:`~pyqtgraph.mkBrush`.
+        """
+        self.brush = fn.mkBrush(*args, **kwargs)
+        if not self.mouseHovering:
+            self.currentBrush = self.brush
+            self.update()
+
+    def setHoverBrush(self, *args, **kwargs):
+        """Set the brush that fills the cursor when hovering over it. Allowable
+        arguments are any that are valid for :func:`~pyqtgraph.mkBrush`.
+        """
+        self.hoverBrush = fn.mkBrush(*args, **kwargs)
+        if self.mouseHovering:
+            self.currentBrush = self.hoverBrush
+            self.update()
+
+    def setPen(self, *args, **kwargs):
+        """Set the pen for drawing the cursor. Allowable arguments are any that
+        are valid for :func:`~pyqtgraph.mkPen`."""
+        self.pen = fn.mkPen(*args, **kwargs)
+        if not self.mouseHovering:
+            self.currentPen = self.pen
+            self.update()
+
+    def setHoverPen(self, *args, **kwargs):
+        """Set the pen for drawing the cursor when hovering over it. Allowable
+        arguments are any that are valid for
+        :func:`~pyqtgraph.mkPen`."""
+        self.hoverPen = fn.mkPen(*args, **kwargs)
+        if self.mouseHovering:
+            self.currentPen = self.hoverPen
+            self.update()
+
+    def setLabel(self, text=None, labelOpts=None):
+        if text is None and self.label is not None:
+            # remove the label if it's already added
+            self.label.scene().removeItem(self.label)
+            self.label = None
         else:
+            if isinstance(text, bool):
+                # convert to default value or empty string
+                text = "x = {:0.3f}, y = {:0.3f}" if text else ""
```

Comment:
In a perfect world, this would do something smart to show e.g. three significant digits for any scale of the plot. But that goes under "future improvements".

Reply:
hmm.... there is some si unit handling used in the spinbox code if I remember right, it's worthy of a look

Reply:
Implemented with 9f182de53197241fcbe09ad873175d9a917d48c1